### DISCUSSION TITLE
RStudio crashes when epinowcast project is loaded

### DISCUSSION THREAD

**Discussion by jamesmbaazam** [epinowcast/epinowcast#398] [Q&A] [answered]
I'm not sure if folks who use R Studio are facing the same issue but whenever I clone the {epinowcast} repository and try opening the `.rproj` or running `devtools::load_all()`, R Studio crashes (See attached screenshot). 

I tried the following, all to no avail:

* Updated R to 4.3.2 from 4.3.1 (didn't work)
* Deleted and recreated `.rproj` file (didn't work)
* Forked the repo (didn't work)
* Re-cloned with write access (didn't work)

💡 What worked for me was downgrading my R Studio version to `2023.06.2-561` (chosen arbitrarily) from `2023.09.1-494`.

I'm guessing it's some weird interaction between the latest R Studio and something in the repo? 

Not sure how to go about troubleshooting, though.

Anyone have any ideas?

![Screenshot 2023-11-23 at 14 30 23](https://github.com/epinowcast/epinowcast/assets/43169674/4b3f47cf-a4db-434f-972a-6a071bf6ae85)

**Comment by seabbs** [comment]
This is very strange indeed. Thanks for reporting @jamesmbaazam. I can't think what this would be. @nikosbosse you saw something similar right? Any light to shed?

**Reply by nikosbosse** [reply]
Yeah I head the same issue, but didn't try to resolve it. I'm currently trying to run this on rstudio.cloud, will report how that goes

**Comment by Bisaloo** [comment]
Minimal reprex:

```r
rlang::check_installed("cmdstanr")
```

I'm not really sure if this issue should be filed in the rlang or RStudio IDE repo :thinking:

**Reply by jamesmbaazam** [reply]
I don't understand.

**Comment by Bisaloo** [comment, marked as answer]
I solved this problem by installing the dailies version of RStudio:

>  rstudio  2023.12.0-daily+345 Ocean Storm (desktop)

**Reply by jamesmbaazam** [reply]
Awesome. Thanks, Hugo. This works for me.

**Reply by seabbs** [reply]
Do we think this issue was on their side or do we think we have an issue of some kind in our configuration that could show up in other edge case settings going forward?

**Reply by Bisaloo** [reply]
It was an issue on their side. An interaction with RStudio & rlang.

As mentioned in https://github.com/epinowcast/epinowcast/discussions/398#discussioncomment-7654127, it was possible to reproduce in a clean RStudio session, completely unrelated to epinowcast, by running `rlang::check_installed("cmdstanr")` and then triggering another rlang action in the console. It's a bit over my head but I'm assuming it's something related to rlang launching a task asynchronously that would then interact with the IDE in some way.

**Reply by jamesmbaazam** [reply]
I wasn't getting anything when I ran that command.

**Reply by Bisaloo** [reply]
It's kind of annoying to reproduce TBH. You have to run that command and then fiddle a little bit with the IDE (I didn't identify the exact trigger). Stuff like starting to type `pak` something worked and crashed the IDE.
But I could consistently get it to crash after fiddling for ~10 sec.

**Reply by jamesmbaazam** [reply]
Interesting. Thanks for helping.
